### PR TITLE
[SDP-962] Change SEP-24 Flow to display different verifications based on Disbursement's verification type

### PR DESCRIPTION
**Pull request by ziyliu**
### What
Modify the `SEP-24` flow to perform verification for an entered phone number based on the latest verification type.

### Why
The current `SEP-24` flow is hardcoded to only accept date of birth but we will have disbursement files that will include pin and national id, and the front-end will need to change to be able to parse those values.

example of the flow using `PIN`
Screenshot 2023-12-14 at 11 01 50 AM

Screenshot 2023-12-14 at 11 05 32 AM

Screenshot 2023-12-14 at 11 06 00 AM

Screenshot 2023-12-14 at 11 06 11 AM

The same tests have been repeated with `DATE_OF_BIRTH` and `NATIONAL_ID`
### Known limitations

[TODO or N/A]

#### PR Structure

* [ ] This PR has reasonably narrow scope (if not, break it down into smaller PRs).
* [ ] This PR does not mix refactoring changes with feature changes (split into two PRs otherwise).
* [ ] This PR's title starts with the name of the package, area, or subject affected by the change.

#### Thoroughness

* [ ] This PR adds tests for the new functionality or fixes.
* [ ] This PR contains the link to the Jira ticket it addresses.

#### Configs and Secrets

* [ ] No new CONFIG variables are required -OR- the new required ones were added to the helmchart's [`values.yaml`] file.
* [ ] No new CONFIG variables are required -OR- the new required ones were added to the deployments ([`pr-preview`], [`dev`], [`demo`], `prd`).
* [ ] No new SECRETS variables are required -OR- the new required ones were mentioned in the helmchart's [`values.yaml`] file.
* [ ] No new SECRETS variables are required -OR- the new required ones were added to the deployments ([`pr-preview secrets`], [`dev secrets`], [`demo secrets`], `prd secrets`).

#### Release

* [ ] This is not a breaking change.
* [ ] **This is ready for production.**. If your PR is not ready for production, please consider opening additional complementary PRs using this one as the base. Only merge this into `develop` or `main` after it's ready for production!

#### Deployment

* [ ] Does the deployment work after merging?

[`values.yaml`]: ../helmchart/sdp/values.yaml
[`pr-preview`]: https://github.com/stellar/kube/blob/d3e4f5dd8aa4c13b45a31a5a937f3e98841171a7/kube001-dev/namespaces/common-previews/stellar-disbursement-platform/backend-helm-values
[`dev`]: https://github.com/stellar/kube/blob/d3e4f5dd8aa4c13b45a31a5a937f3e98841171a7/kube001-dev/namespaces/stellar-disbursement-platform/backend-helm-values
[`demo`]: https://github.com/stellar/kube/blob/d3e4f5dd8aa4c13b45a31a5a937f3e98841171a7/kube001-dev/namespaces/stellar-disbursement-platform/demo/demo-backend-helm-values
[`pr-preview secrets`]: https://github.com/stellar/kube/blob/d3e4f5dd8aa4c13b45a31a5a937f3e98841171a7/kube001-dev/namespaces/common-previews/externalsecrets-common-previews.yaml#L241-L346
[`dev secrets`]: https://github.com/stellar/kube/blob/d3e4f5dd8aa4c13b45a31a5a937f3e98841171a7/kube001-dev/namespaces/stellar-disbursement-platform/stellar-disbursement-platform-externalsecrets.yaml
[`demo secrets`]: https://github.com/stellar/kube/blob/d3e4f5dd8aa4c13b45a31a5a937f3e98841171a7/kube001-dev/namespaces/stellar-disbursement-platform/demo/demo-sdp-externalsecrets.yaml

### CODE REVIEW

File: internal/serve/publicfiles/js/receiver_registration.js [hunk 1]
```diff
@@ -213,7 +231,7 @@ async function submitOtp(event) {
             phone_number: phoneNumber,
             otp: otp,
             verification: verification,
-            verification_type: "date_of_birth",
+            verification_type: verificationField,
```
Comment (ziyliu) [inline internal/serve/publicfiles/js/receiver_registration.js:234]: we should think about making the vocabulary around this more consistent

File: internal/data/receiver_verification.go [hunk 2]
```diff
@@ -106,11 +135,17 @@ func (m ReceiverVerificationModel) Insert(ctx context.Context, sqlExec db.SQLExe
...
 	`
 
-	_, err = sqlExec.ExecContext(ctx, query, verificationInsert.ReceiverID, verificationInsert.VerificationField, hashedValue)
+	if verificationInsert.UpdatedAt != nil {
+		query = fmt.Sprintf(query, ",\nupdated_at\n", ", $4")
+		_, err = sqlExec.ExecContext(ctx, query, verificationInsert.ReceiverID, verificationInsert.VerificationField, hashedValue, *verificationInsert.UpdatedAt)
+	} else {
+		query = fmt.Sprintf(query, "", "")
+		_, err = sqlExec.ExecContext(ctx, query, verificationInsert.ReceiverID, verificationInsert.VerificationField, hashedValue)
+	}
```
Comment (marwen-abid) [inline internal/data/receiver_verification.go:148]: What's the purpose of this change? We shouldn't be updating `updated_at` manually in the main code. 

If the only reason for this change is unit tests, we should find another way to do that. Either directly writing a SQL to update the updated_at field, or calling `UpdateReceiverVerification(..)` that will change the updated_at.

File: internal/serve/publicfiles/js/receiver_registration.js [hunk 3]
```diff
@@ -51,9 +51,9 @@ async function sendSms(phoneNumber, reCAPTCHAToken, onSuccess, onError) {
           recaptcha_token: reCAPTCHAToken,
```
Comment (marwen-abid) [inline internal/serve/publicfiles/js/receiver_registration.js:51]: @quietbits or @ceciliaromao can you help review this part please ? I can't help much with JS. 

@ziyliu did we check with Tori about validation of these fields ? I think there should be some sort of front-end validation for this, especially the PIN.

**Review by marwen-abid — COMMENTED** [review]
This is looking good! Left a few comments

File: internal/serve/publicfiles/js/receiver_registration.js (hunk 3 above)
Comment (quietbits) [reply internal/serve/publicfiles/js/receiver_registration.js:51]: These updates look good to me. :+1:

File: internal/data/receiver_verification.go (hunk 2 above)
Comment (ziyliu) [reply internal/data/receiver_verification.go:148]: ok I see what you're saying, I initially needed this change so that I could create fixtures "created at different times" so that I can test `GetLatestByPhoneNumber`

File: internal/data/receiver_verification.go [hunk 4]
```diff
@@ -91,8 +94,34 @@ func (m ReceiverVerificationModel) GetAllByReceiverId(ctx context.Context, sqlEx
...
+	receiverVerifications := []ReceiverVerification{}
+	query := `
+		SELECT 
+			rv.*
+		FROM 
+			receiver_verifications rv
+		JOIN receivers r ON rv.receiver_id = r.id
+		WHERE r.phone_number = $1
+		ORDER BY
+			rv.updated_at DESC
```
[the remark this answers is not in the document]
Comment (ziyliu) [reply internal/data/receiver_verification.go:108]: great suggestions!

File: internal/serve/publicfiles/js/receiver_registration.js (hunk 3 above)
Comment (ziyliu) [reply internal/serve/publicfiles/js/receiver_registration.js:51]: I forgot to add the backend validation too, but I did just now! I confirmed with Tori when I worked on the disbursement file validations about the requirements.

**Review by marwen-abid — APPROVED** [review]
LGTM!

**Comment by ziyliu** [comment]
Modify the SEP-24 flow to perform verification for an entered phone number based on the latest verification type.

The current SEP-24 flow is hardcoded to only accept date of birth but we will have disbursement files that will include pin and national id, and the front-end will need to change to be able to parse those values.
